src/ASM/ASMbase.C

```diff
@@ -569,6 +569,56 @@ void ASMbase::addLocal2GlobalCpl (int iSlave, int master, const Tensor& Tlg)
 }
 
 
+bool ASMbase::createRgdMasterNode (int& gMaster)
+{
+  bool newNode = gMaster == 0;
+  if (newNode)
+    gMaster = ++gNod;
+  else if (std::find(MLGN.begin()+nnod,MLGN.end(),gMaster) != MLGN.end())
+    return newNode; // This node has already been created
+
+#if SP_DEBUG > 1
+  std::cout <<"Adding extra-ordinary node "<< gMaster
+            <<" for rigid coupling in Patch "<< idx+1 << std::endl;
+#endif
+  myMLGN.push_back(gMaster);
+  myRmaster.insert(myMLGN.size());
+  return newNode;
+}
+
+
+void ASMbase::addRigidMPC (int gSlave, int gMaster, const Vec3& dX)
+{
+  for (unsigned short int dof = 1; dof <= nf; dof++)
+  {
+    MPC* cons = new MPC(gSlave,dof);
+    if (this->addMPC(cons) && cons)
```

Comment:
I don't see how. The MPC object is deleted inside addMPC() method in case it is a duplicate, invalid, etc. Otherwise it is stored in the mpcs map and deleted by the destructor, or the clear() method.

Reply:
i didn't think to check if addMPC deleted the argument. disregard.